src/main/java/seedu/address/logic/commands/suggestions/NusModSuggester.java

```diff
@@ -38,13 +44,21 @@
      * @return Returns a {@link Module} if there is one that matches the {@code moduleCode}.
      */
     protected static Optional<Module> findModuleByCode(final Model model, final String moduleCode) {
-        return model
-                .getModuleList()
-                .asUnmodifiableObservableList()
-                .stream()
-                .filter(module -> {
-                    return module.getModuleCode().toString().equals(moduleCode);
-                }).findFirst();
+        final ModuleCode moduleCodeObj;
+        try {
+            moduleCodeObj = ParserUtil.parseModuleCode(moduleCode);
+        } catch (ParseException e) {
+            return Optional.empty();
+        }
+
+        final AcadYear acadYear = model.getAcadYear();
+        final ModuleId moduleId = new ModuleId(acadYear, moduleCodeObj);
+
+        try {
+            return Optional.of(model.findModule(moduleId));
```

Comment:
Hmm, this method is being called on every keystroke right? Then I think it might not be good to use `model.findModule` since this method will look into cache & api if not in modulelist. I made a small PR #216  to load module list on initialization, then you can do `model.getModuleList().findModule(moduleId)` instead.

Reply:
Noted, will transition to the new API :+1: